tools/sql_extraction/src/main/kotlin/Environment.kt

```diff
@@ -0,0 +1,96 @@
+package com.google.cloud.sqlecosystem.sqlextraction
+
+import com.google.cloud.sqlecosystem.sqlextraction.output.QueryFragment
+
+/**
+ * A mapping from variable name to all possible query fragment values
+ */
+class Environment {
+    private var parentScope: Environment?
+    private var reachingDefs: HashMap<String, QueryFragment?>
+
+    constructor() {
+        this.parentScope = null
+        this.reachingDefs = HashMap()
+    }
```

Comment:
Should we use init block instead?

Reply:
That would execute for the other compiler as well, which is redundant. It might look cleaner though.

Reply:
Update: using init would break any code using the constructor without arguments.